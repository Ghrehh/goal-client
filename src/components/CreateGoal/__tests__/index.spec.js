import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow'
import { mount } from 'enzyme';
import { CreateGoal } from '../';

describe('CreateGoal', () => {
  let testContext = {};

  beforeEach(() => {
    testContext = {};
    testContext.auth = { userId: 1, rememberToken: 'rememberToken' }
    testContext.createGoalMock = jest.fn();

    testContext.component = (
      <CreateGoal
        createGoal={testContext.createGoalMock}
        auth={testContext.auth}
        loading={false}
        error={undefined}
      />
    );
  });

  describe('snapshot', () => {
    beforeEach(() => {
      const renderer = new ShallowRenderer();
      testContext.result = renderer.render(testContext.component);
    });

    it('renders the component correctly', () => {
      expect(testContext.result).toMatchSnapshot();
    });
  });

  describe('submitting form', () => {
    beforeEach(() => {
      const wrapper = mount(testContext.component);
      wrapper.find('.name').instance().value = 'name';
      wrapper.find('button').simulate('click');
    });

    it('calls the create session mock with the correct arguments', () => {
      expect(testContext.createGoalMock).toBeCalledWith({
        variables: { goalName: 'name', auth: testContext.auth }
      });
    });
  });
});
