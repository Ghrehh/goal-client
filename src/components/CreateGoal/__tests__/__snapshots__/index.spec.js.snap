// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateGoal snapshot renders the component correctly 1`] = `
<LoadingAndErrorHandler
  loading={false}
>
  <form>
    <h3>
      Create Goal
    </h3>
    <label>
      Name
      <input
        className="name"
      />
    </label>
    <button
      onClick={[Function]}
    >
      Create Goal
    </button>
  </form>
</LoadingAndErrorHandler>
`;
